src/workerd/jsg/modules.h

```diff
@@ -203,7 +203,13 @@ class ModuleRegistryImpl final: public ModuleRegistry {
     // after first resolve attempt within application has failed, i.e. it is possible for
     // application to override the module.
     // sourceCode has to exist while this ModuleRegistry exists.
-    // The expectation is for this method to be called during the assembly of worker global context.
+    // The expectation is for this method to be called during the assembly of worker global context
+    // after registering all user modules.
+
+    if (entries.find(specifier) != nullptr) {
+      return;
+    }
```

Comment:
Not blocking: I wonder if we may want to add any logging or indication here. I'd like to get a sense of whether anyone is actually encountering this in actual code.